scarb/src/sources/git/client.rs

```diff
@@ -263,20 +279,21 @@ impl<'d> GitCheckout<'d> {
         with_verbosity_flags(&mut cmd, config);
         cmd.args(["--config", "core.autocrlf=false"]);
         cmd.arg("--recurse-submodules");
-        cmd.arg(db.repo.path());
+        let repo = db.repo().await;
+        cmd.arg(repo.path());
```

Comment:
You're locking `repo` until end of the function. Please inline it. If you don't store a variable, but use an object directly in the expression, Rust will `drop` it immediately when expression ends.

Reply:
Closing, as we have decided to go with spawn_blocking instead.